public/angular/js/controllers/incidents/report_incident_modal.js

```diff
@@ -25,9 +25,15 @@ angular.module('Aggie')
       });
 
       modalInstance.result.then(function(report) {
-        report.read = true;     
+        report.read = true;
         Report.update({id: report._id}, report, function(response) {
-          $scope.$parent.r = report;
+          // we're getting the report here because it is easier than
+          // getting the incidents list. we have just updated
+          // _incident, which is an ObjectId. now we want the populated
+          // object again
```

Comment:
This comment doesn't make total sense to me. Please consider revising. What is the purpose here? What is $scope.$parent? Why are we setting its report value? Michal also did some work on streaming updates to report so this stuff may not be necessary.